commit 4973c70a6a4a933ef8d15137be8db90d047781e8
Author: Giulio Eulisse <giulio.eulisse@cern.ch>
Date:   Thu Apr 22 11:49:53 2010 +0200

    Removes GNU extension strndup and replaces it with POSIX compliant code.

diff --git a/lib/rpmchecksig.c b/lib/rpmchecksig.c
index 1b5fca4..34f42e7 100644
--- a/lib/rpmchecksig.c
+++ b/lib/rpmchecksig.c
@@ -593,7 +593,10 @@ static void formatResult(rpmSigTag sigtag, rpmRC sigres, const char *result,
 	if (havekey && (sigres == RPMRC_NOKEY || sigres == RPMRC_NOTTRUSTED)) {
 	    const char *tempKey = strstr(result, "ey ID");
 	    if (tempKey) {
-		char *keyid = strndup(tempKey + 6, 8);
+                char tmp[9];
+                strncpy(tmp, tempKey + 6, 8);
+                tmp[8] = 0;
+                char *keyid = strdup(tmp);
 	    	char *idprob = NULL;
 		rasprintf(&idprob, " %s#%s", signame, keyid);
 		rstrcat(keyprob, idprob);
